src/dsp_tools/commands/excel2json/models/list_node.py

```diff
@@ -0,0 +1,103 @@
+    ) -> ListNode | ListNodeProblem:
+        user_problem = {}
+        if pd.isna(id_):
+            user_problem["name"] = "The name of the node may not be empty."
+        elif isinstance(id_, int | float):
+            id_ = str(id_)
+        elif len(id_) == 0:
+            user_problem["name"] = "The name of the node does not contain any characters."
+        if not labels:
+            user_problem["labels"] = "At least one label per list is required."
+        elif not set(labels).issubset({"en", "de", "fr", "it", "rm"}):
+            user_problem["labels"] = "Only the following languages are supported: 'en', 'de', 'fr', 'it', 'rm'."
+        if user_problem:
+            return ListNodeProblem(id_, user_problem)
+        if not sub_nodes:
+            sub_nodes = []
+        return cls(id_=id_, labels=labels, row_number=row_number, sub_nodes=sub_nodes)
+
+    def to_json(self) -> dict[str, Any]:
+        node = self._make_myself()
+        if self.sub_nodes:
+            node["nodes"] = self._make_subnodes()
+        return node
+
+    def _make_subnodes(self) -> list[dict[str, Any]]:
+        return [nd.to_json() for nd in self.sub_nodes]
+
+    def _make_myself(self) -> dict[str, Any]:
+        return {"name": self.id_, "labels": self.labels}
+
+
+@dataclass
+class ListRoot:
+    id_: str
+    labels: dict[str, str]
+    nodes: list[ListNode] = field(default_factory=list)
+    comments: dict[str, str] | None = None
+
+    @classmethod
+    def create(
+        cls,
+        id_: str,
+        labels: dict[str, str],
+        nodes: list[ListNode],
+        comments: dict[str, str] | None = None,
+    ) -> ListRoot | ListProblem:
+        user_problem = {}
+        if pd.isna(id_):
+            user_problem["name"] = "The name of the list may not be empty."
+        elif isinstance(id_, int | float):
+            id_ = str(id_)
+        elif not isinstance(id_, str):
+            user_problem["name"] = "The name of the list is not a string."
+        elif len(id_) == 0:
+            user_problem["name"] = "The name of the list does not contain any characters."
+        if not labels:
+            user_problem["labels"] = "At least one label per list is required."
+        elif not set(labels).issubset({"en", "de", "fr", "it", "rm"}):
+            user_problem["labels"] = "Only the following languages are supported: 'en', 'de', 'fr', 'it', 'rm'."
+        if comments:
+            if not set(comments).issubset({"en", "de", "fr", "it", "rm"}):
+                user_problem["comments"] = "Only the following languages are supported: 'en', 'de', 'fr', 'it', 'rm'."
+        if len(nodes) == 0:
+            user_problem["list nodes"] = "At least one node per list is required."
+        if user_problem:
+            return ListProblem(id_, user_problem)
+        return cls(id_=id_, labels=labels, comments=comments, nodes=nodes)
+
+    def to_json(self) -> dict[str, Any]:
```

Comment:
Perhaps, `as_dict` would be more precise, because AFAIU, JSON is the serialized string, whereas the thing that you produce is a python dictionary, which has nothing to do with a particular string format like JSON.